eng/Compilers.props

```diff
@@ -6,7 +6,17 @@
     <CompilerVersion>3.6.0-2.20166.2</CompilerVersion>
   </PropertyGroup>
 
+  <PropertyGroup Condition="!($(MSBuildProjectName.EndsWith('Tests', StringComparison.OrdinalIgnoreCase)))">
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn></NoWarn> <!-- Make sure the list is empty - by default it is not -->
```

Comment:
I would just remove this as I don't think it is a good idea to hard reset it to empty here. There are some warnings that we may get coming from Arcade(the central infrastructure used in dotnet repos) which may be expected so we don't want to shoot ourselves in the foot here and remove those.

Reply:
The problem is that the default list we inherit contains CS1591, which we explicitly _do not_ want to disable. So without this hard reset, the change basically does not work.

Reply:
`<NoWarn>$(NoWarn.Replace("1591", ""))</NoWarn>` does the trick, it seems.

Reply:
😆 feels hacky but I guess it's ok if it works. The only last thing that makes me uncomfortable is that now we depend on the import order, as we are depending on Compilers.props being imported after the NoWarn gets set, but I guess this is fine for now as whenever we break this it will be easy to diagnose as it will break the build.